### PR TITLE
Migrate `src/utils/dom` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**
- Migrates to TypeScript
- Removes unused features from `createElement` and `hasNextNodeInDocumentOrder`

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
Waiting for #2821

File: src/utils/dom.ts [hunk 1]
```diff
@@ -93,23 +81,23 @@ export function injectStyleSheet(doc, styleText) {
...
  *  b. Any of the element ancestors has a nextSibling.
- * @param {!Element} element
- * @param {?Node=} stopNode
- * @return {boolean}
  */
-export function hasNextNodeInDocumentOrder(element, stopNode) {
-  let currentElement = element;
+export function hasNextNodeInDocumentOrder(
+  element: HTMLElement,
+  stopNode: (Node | null) | undefined
```
Comment (fikus) [inline src/utils/dom.ts:91]: Can this be typed as `Node?`?

File: src/utils/dom.ts [hunk 2]
```diff
@@ -93,23 +81,23 @@ export function injectStyleSheet(doc, styleText) {
...
+): boolean {
+  let currentElement: HTMLElement | null = element;
   do {
     if (currentElement.nextSibling) {
       return true;
     }
   } while (
-    (currentElement = currentElement.parentNode) &&
-    currentElement != stopNode
+    (currentElement = currentElement.parentElement) &&
```
Comment (fikus) [inline src/utils/dom.ts:99]: Does this behavior need to be changed?

File: src/utils/dom.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ *  a. The node itself has a nextSibling.
+ *  b. Any of the node ancestors has a nextSibling.
+ */
+export function hasNextNodeInDocumentOrder(node: Node): boolean {
+  let currentNode: Node | null = node;
+  do {
+    if (currentNode.nextSibling) {
+      return true;
+    }
+  } while ((currentNode = currentNode.parentNode));
```
Comment (fikus) [inline src/utils/dom.ts:95]: nit: redundant parentheses.

**Review by fikus — APPROVED** [review]

File: src/utils/dom.ts (hunk 3 above)
Comment (ChrisAntaki) [reply src/utils/dom.ts:95]: Prettier wills it

**Pull request merged**